backends/p4tools/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -0,0 +1,949 @@
+#include <algorithm>
+#include <cstddef>
+#include <functional>
+#include <ostream>
+#include <vector>
+
+#include <boost/none.hpp>
+
+#include "backends/p4tools/common/lib/formulae.h"
+#include "backends/p4tools/common/lib/ir.h"
+#include "backends/p4tools/common/lib/symbolic_env.h"
+#include "backends/p4tools/common/lib/trace_events.h"
+#include "ir/ir.h"
+#include "lib/cstring.h"
+#include "lib/exceptions.h"
+#include "lib/safe_vector.h"
+
+#include "backends/p4tools/testgen//lib/exceptions.h"
+#include "backends/p4tools/testgen/core/externs.h"
+#include "backends/p4tools/testgen/core/program_info.h"
+#include "backends/p4tools/testgen/core/small_step/expr_stepper.h"
+#include "backends/p4tools/testgen/core/small_step/small_step.h"
+#include "backends/p4tools/testgen/lib/continuation.h"
+#include "backends/p4tools/testgen/lib/execution_state.h"
+
+namespace P4Tools {
+
+namespace P4Testgen {
+
+void ExprStepper::setFields(ExecutionState* nextState,
```

Comment:
maybe this file should be called core_externs_stepper.

Reply:
It also contains internal externs. Might make sense to split things up.